build/LocalizationTasks/EmitLocalizedResources.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.Build.LocalizationTasks
+{
+    public class EmitLocalizedResources : Task
+    {
+        private static readonly string NeutralResxMetadata = "NeutralResx";
+        private static readonly string ParentXlfMetadata = "ParentXlf";
+        private static readonly string ChildResxMetadata = "ChildResx";
+        private static readonly string ComputedCultureCodeMetadata = "ComputedCulture";
+        private static readonly string LogicalNameMetadata = "LogicalName";
+
+        [Required]
+        public ITaskItem[] NeutralResources { get; set; }
+
+        /// <summary>
+        /// Name of the main assembly
+        /// </summary>
+        [Required]
+        public string AssemblyName { get; set; }
+        
+        /// <summary>
+        /// Directory path where localized resx should be generated in 
+        /// </summary>
+        [Required]
+        public string LocalizedResxRoot { get; set; }
+
+        /// <summary>
+        /// Path to the root directory that contains the xlf files for a neutral resource
+        /// Assumes all neutral resource have the same relative path to their xlf
+        /// </summary>
+        [Required]
+        public string RelativePathToXlfRoot { get; set; }
+
+        [Output]
+        public ITaskItem[] ResolvedXlfResources { get; set; }
+
+        [Output]
+        public ITaskItem[] ResolvedLocalizedResxResources { get; set; }
+
+        public override bool Execute()
+        {
+            if (!NeutralResources.Any())
+            {
+                Log.LogMessage("No neutral resources found");
+                return true;
+            }
+
+            ResolvedXlfResources = NeutralResources.SelectMany(ComputeXlfResources).ToArray();
+
+            ResolvedLocalizedResxResources = ResolvedXlfResources.Select(ComputeLocalizedResource).ToArray();
+
+            return true;
```

Comment:
MSBuild idiom: always return `!Log.HasLoggedErrors`--this prevents situations where you fail the build (`return false`) but never logged an error (so there's nothing to go on when it happens and your build fails).